src/OpenTelemetry/Metrics/MetricTypeExtensions.cs

```diff
@@ -59,7 +59,7 @@ public static bool IsGauge(this MetricType self)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static bool IsHistogram(this MetricType self)
         {
-            return (self & METRIC_TYPE_MASK) == METRIC_TYPE_HISTOGRAM;
+            return (self & METRIC_TYPE_HISTOGRAM) == METRIC_TYPE_HISTOGRAM;
```

Comment:
`(self & METRIC_TYPE_HISTOGRAM) == METRIC_TYPE_HISTOGRAM` is misleading, use `self.HasFlag(METRIC_TYPE_HISTOGRAM)`.